internal/pkg/transformation/kubernetes.go

```diff
@@ -279,8 +299,41 @@ func (p *PodMapper) toDeviceToPod(
 
 	for _, pod := range devicePods.GetPodResources() {
 		for _, container := range pod.GetContainers() {
-			for _, device := range container.GetDevices() {
+			podInfo := PodInfo{
+				Name:      pod.GetName(),
+				Namespace: pod.GetNamespace(),
+				Container: container.GetName(),
+			}
+
+			if dynamicResources := container.GetDynamicResources(); len(dynamicResources) > 0 {
+				for _, dr := range dynamicResources {
+					for _, claimResource := range dr.GetClaimResources() {
```

Comment:
As I look through this rather briefly, I notice that here we have four nested for loops and the outest loop has a huge body (in terms of vertical length). I hope we can improve this, especially in terms of readability :) Maybe we can make the body of the outest loop really short, so that all that remains is an expressive scaffold that calls into well-named functions.